File: ui/src/main/java/com/peakmain/ui/utils/database/IDaoSupport.java
package com.peakmain.ui.utils.database;

import android.database.sqlite.SQLiteDatabase;

import java.util.List;

public interface IDaoSupport<T> {
    void init(SQLiteDatabase sqLiteDatabase, Class<T> clazz);

    //插入数据
    long insert(T t);

    //批量插入
    void insert(List<T> datas);

    QuerySupport<T> querySupport();

    int delete(String whereClause, String... whereArgs);

    int update(T obj, String whereClause, String... whereArgs);

}

File: ui/src/main/java/com/peakmain/ui/utils/database/IDaoSupport.kt
package com.peakmain.ui.utils.database

import android.database.sqlite.SQLiteDatabase

interface IDaoSupport<T> {
    fun init(sqLiteDatabase: SQLiteDatabase?, clazz: Class<T>?)

    //插入数据
    fun insert(t: T): Long

    //批量插入
    fun insert(datas: List<T>?)
    fun querySupport(): QuerySupport<T>?
    fun delete(whereClause: String?, vararg whereArgs: String?): Int
    fun update(obj: T, whereClause: String?, vararg whereArgs: String?): Int
}